lib/graphql/schema/printer.rb

```diff
@@ -56,12 +64,24 @@ def print_type(type)
         TypeKindPrinters::STRATEGIES.fetch(type.kind).print(type)
       end
 
+      def print_directive(directive)
+        TypeKindPrinters::DirectivePrinter.print(directive)
+      end
+
       module TypeKindPrinters
-        module FieldPrinter
-          def print_fields(type)
-            type.all_fields.map{ |field| "  #{field.name}#{print_args(field)}: #{field.type}" }.join("\n")
+        module DeprecatedPrinter
+          def print_deprecated(field_or_enum_value)
+            return unless field_or_enum_value.deprecation_reason
+
+            if field_or_enum_value.deprecation_reason == ''
+              ' @deprecated'
+            else
+              " @deprecated(reason: #{field_or_enum_value.deprecation_reason.to_s.inspect})"
+            end
```

Comment:
Good catch. The code is clearer with the `case` statement. I'll make that change.